internal/characters/xianyun/plunge.go

```diff
@@ -0,0 +1,251 @@
+	lowPlungeFramesXY[action.ActionSwap] = 48
+}
+
+func (c *char) HighPlungeAttack(p map[string]int) (action.Info, error) {
+	defer c.Core.Player.SetAirborne(player.Grounded)
+
+	// last action must be skill (for leap)
+	// dont need to check airborne for this because she can plunge if she's on the ground anyways
+	if c.StatusIsActive(skillStateKey) {
+		return c.driftcloudWave()
+	}
+
+	switch c.Core.Player.Airborne() {
+	case player.AirborneVenti:
+		return action.Info{}, fmt.Errorf("%s high_plunge while airborne due to venti is unimplemented due to lack of frame data. Please see https://docs.gcsim.app/mechanics/frames for how to contribute", c.Base.Key.String())
+	case player.AirborneXianyun:
+		return c.highPlungeXY(p)
+	default:
+		return action.Info{}, fmt.Errorf("%s high_plunge can only be used while airborne", c.Base.Key.String())
+	}
+}
+
+func (c *char) driftcloudWave() (action.Info, error) {
+	skillInd := c.skillCounter - 1
+	skillArea := combat.NewCircleHitOnTarget(c.Core.Combat.Player(), nil, plungeRadius[skillInd])
+	skillHitmark := plungeHitmarks[skillInd]
+	c.QueueCharTask(func() {
+		ai := combat.AttackInfo{
+			ActorIndex: c.Index,
+			Abil:       fmt.Sprintf("Driftcloud Wave %v", c.skillCounter),
+			AttackTag:  attacks.AttackTagPlunge,
+			ICDTag:     attacks.ICDTagNone,
+			ICDGroup:   attacks.ICDGroupDefault,
+			StrikeType: attacks.StrikeTypeDefault,
+			Element:    attributes.Anemo,
+			Durability: 25,
+			Mult:       leap[skillInd][c.TalentLvlSkill()],
+		}
+		snap := c.Snapshot(&ai)
+		c.c6mod(&snap)
+
+		c.Core.QueueAttackWithSnap(ai, snap, skillArea, 0, c.particleCB(), c.a1cb(), c.c4cb())
+
+		// reset window after leap
+		c.DeleteStatus(skillStateKey)
+		c.skillCounter = 0
+		c.skillEnemiesHit = nil
+		c.skillSrc = noSrcVal
+	}, skillHitmark)
+
+	return action.Info{
+		Frames:          frames.NewAbilFunc(driftcloudFrames[skillInd]),
+		State:           action.PlungeAttackState,
+		AnimationLength: driftcloudFrames[skillInd][action.InvalidAction],
+		CanQueueAfter:   driftcloudFrames[skillInd][action.ActionSwap],
+	}, nil
+}
+
+// Low Plunge attack damage queue generator
+// Use the "collision" optional argument if you want to do a falling hit on the way down
+// Default = 0
+func (c *char) LowPlungeAttack(p map[string]int) (action.Info, error) {
+	defer c.Core.Player.SetAirborne(player.Grounded)
+
+	// last action must be skill (for leap)
+	// dont need to check airborne for this because she can plunge if she's on the ground anyways
+	if c.StatusIsActive(skillStateKey) {
+		return c.driftcloudWave()
+	}
+
+	switch c.Core.Player.Airborne() {
+	case player.AirborneXianyun:
+		return c.lowPlungeXY(p)
+	default:
+		return action.Info{}, fmt.Errorf("%s low_plunge can only be used while airborne", c.Base.Key.String())
+	}
+}
+
+func (c *char) lowPlungeXY(p map[string]int) (action.Info, error) {
+	if c.Core.Player.CurrentState() != action.JumpState {
+		return action.Info{}, errors.New("only plunge after using jump")
+	}
```

Comment:
This is unnecessary, please remove it. If Airborne, then previous action was definitely jump.